src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -716,14 +716,15 @@ public void close() {
 		 */
 		private CompletableFuture<Void> receivePumpAsync() {
 			return receivePumpCoreAsync().handle((keepGoing, ex) -> {
-				if (keepGoing) {
+				if (keepGoing) {				
 					receivePumpAsync();
```

Comment:
>receivePumpAsync [](start = 5, length = 16)

Is this truly supposed to fall through and call onOffline?

Reply:
isn't supposed to?  Exit keepGoing because it's going offline

---
In reply to: [257831842](https://github.com/Azure/azure-relay-java/pull/16#discussion_r257831842) [](ancestors = 257831842)

Reply:
The thing is line 720 kicks off a reconnect, but doesn't wait for it.

---
In reply to: [257831842](https://github.com/Azure/azure-relay-java/pull/16#discussion_r257831842) [](ancestors = 257831842)

Reply:
You are absolutely right. Having the else block down there should fix it

---
In reply to: [258567226](https://github.com/Azure/azure-relay-java/pull/16#discussion_r258567226) [](ancestors = 258567226,257831842)